src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,277 @@
+            return 1.0
+        if backend is None:  # pragma: no cover
+            backend = GlobalBackend()
+
+        result = backend.execute_circuit(circuit, nshots=nshots)
+        observable = _get_observable(j, nqubits)
+        return result.expectation_from_samples(observable)
+
+
+def _gate_tomography(
+    nqubits=None,
+    gate=None,
+    nshots=int(1e4),
+    invert_register=None,
+    noise_model=None,
+    backend=None,
+):
+    """Runs gate set tomography for a 1 or 2 qubit gate.
+
+    Args:
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        gate (:class:`qibo.gates.abstract.Gate`, optional): The gate to perform gate set tomography on. If gate=None, then gate set tomography will be performed for an empty circuit.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied to simulate noisy computation.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+    Returns:
+        ndarray: array with elements ``jk`` equivalent to either :math:`\\text{tr}(Q_{j} \\, \\rho_{k})` or :math:`\\text{tr}(Q_{j} \\, O_{l} \\rho_{k})` where :math:`O_{l}` is the l-th operation in the original circuit.
+    """
+
+    # Check if gate is 1 or 2 qubit gate.
+    if not nqubits in (1, 2):
+        raise_error(
+            ValueError,
+            f"nqubits given as {nqubits}. nqubits needs to be either 1 or 2.",
+        )
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if gate is not None:
+        if nqubits != len(gate.qubits):
+            raise_error(
+                ValueError,
+                f"Mismatched inputs: nqubits given as {nqubits}. {gate} is a {len(gate.qubits)}-qubit gate.",
+            )
+        gate = gate.__class__(*gate.qubits, **gate.init_kwargs)
+
+    # GST for empty circuit or with gates
+    matrix_jk = np.zeros((4**nqubits, 4**nqubits))
+    for k in range(4**nqubits):
+        circ = Circuit(nqubits, density_matrix=True)
+        circ.add(_prepare_state(k, nqubits))
+        if invert_register is not None:
+            inverted_circuit = reset_register(circ, invert_register)
+            circ.add(inverted_circuit.on_qubits(*invert_register))
+
+        if gate is not None:
+            circ.add(gate)
+
+        for j in range(4**nqubits):
+            new_circ = circ.copy()
+            measurements = _measurement_basis(j, nqubits)
+            new_circ.add(measurements)
+            if noise_model is not None and backend.name != "qibolab":
+                new_circ = noise_model.apply(new_circ)
+            expectation_val = _expectation_value(
+                new_circ, k, j, nshots, backend=backend
+            )
+            matrix_jk[j, k] = expectation_val
+    return matrix_jk
+
+
+def GST(
+    gate_set: Union[tuple, set, list],
+    nshots: int = int(1e4),
+    noise_model=None,
+    include_empty: bool = False,
+    invert_register: tuple = None,
+    backend=None,
```

Comment:
I think we should add to this function the option to have the matrices to be returned in the Pauli-Liouville representation directly.

Reply:
I agree, working on it now.